pkg/tfsandbox/state.go

```diff
@@ -23,23 +23,46 @@ import (
 )
 
 const defaultStateFile = "terraform.tfstate"
+const defaultLockFile = ".terraform.lock.hcl"
 
-func (t *Tofu) PullState(_ context.Context) (json.RawMessage, bool, error) {
+func (t *Tofu) PullStateAndLockFile(_ context.Context) (json.RawMessage, []byte, error) {
```

Comment:
DoXAndY makes for a bit of an awkward API. The call sites could do this separately. 

Perhaps a quick comment here that if there is no Terraform lock file found, the method will return `nil []byte` without an error.

Reply:
> DoXAndY makes for a bit of an awkward API. The call sites could do this separately.

I thought about this, but since you always want to do both I though splitting them up and requiring the user to know that they always have to us both is kind of awkward.

I could rename these functions to be more generic. Something like `PullWorkspace` and `PushWorkspace`? What do you think about that?

> Perhaps a quick comment here that if there is no Terraform lock file found, the method will return nil []byte without an error.

👍🏻

Reply:
I'd leave as is then. 

state push/pull are commands: https://opentofu.org/docs/cli/commands/state/pull/ - although not implemented that way yet, the names are definitely referring to that concept. 

Workspace is not a concept. Well it is in https://opentofu.org/docs/language/state/workspaces/ but that does not match the use here so that would be confusing.